File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int ans = -1;
        unordered_map<int,int>freqmap;
        for(int i=0;i<nums.size();i++)
        {
            freqmap[nums[i]]++;
            if(freqmap[nums[i]]==2)
                return nums[i];
        }
        return ans;
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        
        int n = nums.length;
        int[] countArr = new int[n];
        int dup = 0;
        
        
        for(int i=0;i<n;i++)
        {
            if(countArr[nums[i]]<0)
            {
                 dup = nums[i];
                 break;
            }
            else
                countArr[nums[i]]--;
            
        }
        return dup;
    }
}